commons-numbers-complex-arrays/src/test/java/org/apache/commons/numbers/complex/arrays/ComplexListTest.java

```diff
@@ -105,13 +116,31 @@ void testAddAndAddAll() {
         assertListOperation(list -> list.addAll(list1), l5, l6);
 
         //Test for adding an empty list to an empty list
-        ComplexList list = new ComplexList();
+        ComplexList list = ComplexList.interleaved();
         assertListOperation(l -> {
             l.addAll(list);
             return l.addAll(0, list);
         });
     }
 
+    @Test
+    void testSetAddAndAddAllNullPointerException() {
+        ComplexList list1 = generateComplexList(3);
+        Assertions.assertThrows(NullPointerException.class, () -> list1.add(null));
+        Assertions.assertThrows(NullPointerException.class, () -> list1.add(0, null));
+        Assertions.assertThrows(NullPointerException.class, () -> list1.set(1, null));
+
+        List<Complex> list2 = generateList(3);
+        for (int i = 0; i < list2.size(); i++) {
```

Comment:
This loop is setting all elements to null. Just set one of them:
`list2.set(1, null)`